packages/core/content-manager/server/src/history/services/history.ts

```diff
@@ -43,19 +79,38 @@ const createHistoryService = ({ strapi }: { strapi: LoadedStrapi }) => {
        * TODO: Fix the types for the middleware
        */
       strapi.documents.use(async (context, next) => {
-        // @ts-expect-error ContentType is not typed correctly on the context
-        const contentTypeUid = context.contentType.uid;
-        const params = context.args.at(-1) as any;
         // Ignore actions that don't mutate documents
-        if (!['create', 'update', 'publish', 'unpublish'].includes(context.action)) {
+        if (
+          !['create', 'update', 'publish', 'unpublish', 'discardDraft'].includes(context.action)
```

Comment:
you might want to cover the case where multiple locales are published/unpublished/discarded (using locale="*" param)

Reply:
ok, in that case would the versions array contain several documents?

Reply:
As it turns out we only need to support requests coming from the CM. Do you know if it is / will be possible to publish in several locales at once from the CM?

I think we will also need to handle this case for bulk publish/unpublish for a single locale as well but I prefer to set that up in another PR when bulk actions are working in v5.

Reply:
And we will need to filter out the requests that do not come from the content manager.

Reply:
>. Do you know if it is / will be possible to publish in several locales at once from the CM?
Yep! we will support that option

Reply:
Thanks!! I've created another ticket to handle the bulk action stuff in a follow up PR once all that's ready in the content manager.